### DISCUSSION TITLE
Using Web COmponents in React Framework

### DISCUSSION THREAD

**Discussion by gkamlesh21** [storybookjs/storybook#18191] [Help] [answered]
Hello Team, We have created a library using web components framework in storybook UI. Our purpose is to server that library in React and Angular. Is that possible to use cross framework in storybook UI.
----------------------
Example.
AtomicButton.js
import { html } from 'lit-html';
import { styleMap } from 'lit-html/directives/style-map.js';
import './button.scss';

/**
 * Primary UI component for user interaction
 */
export const AtomicButton = ({ backgroundColor = null, type, label, onClick, disabled = false, fullWidth}) => {
  const mode = `storybook-button--${type || "primary"}`;
  return html`
    <button
      type="button"
      class="storybook-button ${mode} ${mode}${disabled ? `--disabled` : ''} ${fullWidth ? `fullWidth` :''}"
      style=${styleMap({ backgroundColor })}
      @click=${onClick}
    >
      ${label}
    </button>
  `;
};

export default AtomicButton;

-------------------------------
//AtomicButton.stories.js
import { AtomicButton } from './AtomicButton'

// More on default export: https://storybook.js.org/docs/web-components/writing-stories/introduction#default-export
export default {
  title: 'Atoms/Button',
  // More on argTypes: https://storybook.js.org/docs/web-components/api/argtypes
  argTypes: {
    backgroundColor: { control: 'color' },
    onClick: { action: 'onClick' },
    type: {
      control: { type: 'select' },
      options: ['primary', 'primarySecondary', 'secondary'],
    },
  },
};

// More on component templates: https://storybook.js.org/docs/web-components/writing-stories/introduction#using-args
const Template = (args) => AtomicButton(args);

export const Primary = Template.bind({});
// More on args: https://storybook.js.org/docs/web-components/writing-stories/args
Primary.args = {
  type: "primary",
  disabled: false,
  fullWidth:false,
  label: 'Primary Action',
};

---------------------
//In React Framework - App.js
![image](https://user-images.githubusercontent.com/30630929/167821253-371a914b-af39-44d2-9bef-51564d69b63c.png)

**Comment by break-stuff** [comment, marked as answer]
You would have to setup separate instances for each framework, but you should be able you use Storybook composition to help with this.

https://storybook.js.org/docs/sharing/storybook-composition

**Reply by joevaugh4n** [reply]
Thank you for jumping in here, @break-stuff!
